### PR TITLE
Specify when record struct synthesized members are 'readonly'

### PR DESCRIPTION
**Pull request by RikkiGibson**

### CODE REVIEW

**Comment by jcouv** [comment]
I must have gotten confused. The note that I'd taken to test plan was this:
"LDM: Emitting GetHashCode and Equals as readonly (conditionally) (answer: let's not do anything special)"

Does that differ from LDM notes?

**Review by Youssef1313 — COMMENTED** [review]
Can't `PrintMembers` and `Deconstruct` be always readonly? I don't get why there is a restriction. Those methods aren't supposed to write to fields/properties.
Anything I'm missing?

**Comment by svick** [comment]
@Youssef1313 I believe the problematic case is when you have a property with a non-readonly getter.

File: proposals/record-structs.md [hunk 1]
```diff
@@ -143,6 +143,8 @@ The record struct includes a synthesized method equivalent to a method declared
 public override string ToString();
 ```
 
+If the record's printable members do not include a readable property with a non-`readonly` `get` accessor, then the synthesized `PrintMembers` and `ToString` methods are `readonly`.
```
Comment (jcouv) [inline proposals/record-structs.md:146]: The situation where a printable member is a non-`readonly` field should also be accounted for.

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 1)

File: proposals/record-structs.md (hunk 1 above)
Comment (RikkiGibson) [reply proposals/record-structs.md:146]:
```suggestion
If the record's printable members do not include a readable property with a non-`readonly` `get` accessor, then the synthesized `PrintMembers` and `ToString` methods are `readonly`. There is no requirement for the record's fields to be `readonly` for the `PrintMembers` and `ToString` methods to be `readonly`.
```

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by jcouv** [comment]
Do we want to hold off this PR until we're ready to make the corresponding implementation change?

**Comment by RikkiGibson** [comment]
> Do we want to hold off this PR until we're ready to make the corresponding implementation change?

Yeah, I think we should just merge it in sync with the implementation change.

**Comment by RikkiGibson** [comment]
It was a bogus conflict (GitHub found a conflict, but local git reported no conflicts upon merge).

**Pull request merged**